virtlauncher, converter_test: use static domxml testdata

dankenigsberg opened the pull request:
This PR tidies up how balloon device converter tests are written. Instead of computing the expected result during the test, tesdata is made into static xmls. Several tests that need low level modification of specific fields are moved into balloon_test.go, where the expected value are embedded into the test code.

```release-note
NONE
```

dankenigsberg committed:
launcher, converter_test: localize stats-period unit tests

Move 6 stats-period unit tests that computed their expected domxml
template at test time from converter_test.go into the focused
balloon_test.go, where the expected state is stated explicitly.

Simplify to two tests: one for non-zero period and one for zero,
since BalloonDomainConfigurator is architecture-agnostic. Include all
BalloonDomainConfigurator options explicitly in each test.

Assisted-By: Claude Sonnet 4.6
Signed-off-by: Dan Kenigsberg <danken@redhat.com>

dankenigsberg committed:
localize tests for AutoAttacheMemBalloon=False

It is not clear to me why we had a test per architecture. I suspect that
it was an artifact of how the tesdata was created. A single test is
enough.

Signed-off-by: Dan Kenigsberg <danken@redhat.com>

dasionov commented:
/lgtm

orelmisan commented on pkg/virt-launcher/virtwrap/converter/compute/balloon_test.go:
```diff
@@ -94,4 +94,162 @@ var _ = Describe("Balloon Domain Configurator", func() {
 	})
+
+	Context("with memballoon stats period", func() {
+		It("should configure memballoon with stats period 5 for amd64", func() {
```
Given that `BalloonDomainConfigurator` is architecture-agnostic, I suggest we simplify the tests by only covering period = 0 and period != 0 cases.

dankenigsberg replied:
It took me awhile, but I believe I've addressed your comments.

orelmisan reviewed (changes requested):
Thank you for simplifying our converter @dankenigsberg!

dankenigsberg committed:
launcher, converter_test: use static testdata

Embed test-time computed value in the hard-coded domxmls used in the
test. Tests should be simple and straight-forward, using as little
compute power as possible.

Also adapt balloon_test.go to the updated BalloonDomainConfigurator API
that replaced BalloonWithArchitecture+BalloonWithUseVirtioTransitional
with a single BalloonWithVirtioModel option.

Assisted-By: Claude Sonnet 4.6
Signed-off-by: Dan Kenigsberg <danken@redhat.com>

dankenigsberg committed:
balloon_test: fix s390x PV test using SEV-configured VMI

The s390x PV test was seemingly written by copy-pasting the amd64 SEV test and
not updating the VMI setup. The amd64 test creates a VMI with AMD SEV
launch security configured which does not exist on s390x:

  vmi := libvmi.New(withLaunchSecurity(v1.LaunchSecurity{SEV: &v1.SEV{}}))

The configurator's Configure() method does not read the VMI's launch
security spec - it acts only on the useLaunchSecuritySEV and
useLaunchSecurityPV flags passed to NewBalloonDomainConfigurator — so
the test passed despite the mistake. Still, a test for the s390x PV
path has no business presenting a VMI with AMD SEV configured. Fix it
to use a plain VMI.

The copy-paste origin: commit 873e6eb886 ("launcher, converter:
Introduce BalloonDomainConfigurator") introduced both the amd64 SEV
and the s390x PV test at once. The s390x block was copied from the
amd64 block with BalloonWithUseLaunchSecuritySEV flipped to false and
BalloonWithUseLaunchSecurityPV flipped to true, but the VMI line was
left as-is. A later commit (591dae19bc, "launcher, converter: Make
Balloon configurator arch-independent") fixed the wrong virtio model
in that test but left the VMI untouched.

Assisted-By: Claude Sonnet 4.6
Signed-off-by: Dan Kenigsberg <danken@redhat.com>

orelmisan commented:
/test pull-kubevirt-unit-test

orelmisan commented on pkg/virt-launcher/virtwrap/converter/compute/balloon_test.go:
```diff
@@ -152,4 +152,32 @@ var _ = Describe("Balloon Domain Configurator", func() {
+			var domain api.Domain
+
+			configurator := compute.NewBalloonDomainConfigurator(
+				compute.BalloonWithArchitecture(arch),
```
nit: `BalloonWithArchitecture` doesn't exist. This commit doesn't compile on its own - it's fixed by the next commit  which switches to `BalloonWithVirtioModel`.
Consider using `BalloonWithVirtioModel` directly.

orelmisan commented on pkg/virt-launcher/virtwrap/converter/compute/balloon_test.go:
```diff
@@ -175,9 +175,9 @@ var _ = Describe("Balloon Domain Configurator", func() {
-			Entry("for amd64", "amd64"),
-			Entry("for arm64", "arm64"),
-			Entry("for s390x", "s390x"),
+			Entry("for amd64", "virtio-non-transitional"),
+			Entry("for arm64", "virtio-non-transitional"),
```
Since the balloon configurator is architecture-agnostic, the "for amd64" and "for arm64" entries are identical - same input, same code path, same expected output. 
One of them can be dropped.

orelmisan reviewed (approved):
Thank you @dankenigsberg!
Approving for SIG compute
/approve

enp0s3 reviewed (commented):
/approve

orelmisan commented:
/retest-required

orelmisan commented:
/unhold

orelmisan commented:
CI is green
/unhold
